doc/nrf/gs_programming.rst

```diff
@@ -119,3 +119,78 @@ After completing the :ref:`manual <build_environment_cli>` or :ref:`automatic <g
+
+.. _gs_programming_output:
+
+Output build files
+******************
+
+The building process produces each time an *image file*.
+
+.. output_build_files_info_start
+
+The image file, also referred to as *executable*, *program*, or *elf file*, consists of pieces of code and data that are identified by image-unique names recorded in a single symbol table.
+The symbol table exists as metadata in a :file:`.elf` or :file:`.exe` file and is not included when the image is converted to a hexidecimal file for programming.
+Instead, a linker places the code and data at their addresses.
+
+Only images require this linking process.
+Object files do not require linking.
+Depending on the application and the SoC, you can use one or several images.
+To determine if you have zero, one, or more images, count the number of times the linker runs.
+
+.. output_build_files_info_end
+
+.. output_build_files_table_start
+
+The following table lists build files that can be generated as output when building firmware for supported :ref:`build targets <app_boards>`.
+The table includes files for single-core and multi-core programming scenarios for both |VSC| and command line building methods.
+
++---------------------------------+-------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------+
+| File                            | File format                                                                                     | Programming scenario                                                     |
++=================================+=================================================================================================+==========================================================================+
+| :file:`zephyr.hex`              | Default full image.                                                                             | * Programming non-secure domain (``_ns``) or single-core build targets.  |
+|                                 | In a multi-image build, several :file:`zephyr.hex` are generated, one for each image.           | * Testing DFU procedure with nrfjprog.                                   |
++---------------------------------+-------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------+
+| :file:`merged.hex`              | The result of merging all :file:`zephyr.hex` files for all images for a core                    | * Programming multi-core application.                                    |
+|                                 | in a multi-image build. Not used by Nordic Semiconductor's build targets.                       | * Testing DFU procedure with nrfjprog.                                   |
```

Comment:
Not used by Nordic Semiconductor's build targets? Did you mean "Not used by single core build targets"?

Reply:
From what I understood, it's not used at all, but rather used to create merged.hex... Is that incorrect?

Reply:
But it is a description of `merged.hex`. I can tell that if you build a multi-image application for nRF52840 it is used when you run `west flash`.

Reply:
I meant, merged_domain.hex, sorry...

Reply:
Should be clear now, @Damian-Nordic , after some clarifications from Haakon...